xbmc/music/MusicInfoLoader.cpp

```diff
@@ -102,19 +101,21 @@ bool CMusicInfoLoader::LoadAdditionalTagInfo(CFileItem* pItem)
     if (database.GetAlbum(param.GetAlbumId(), album, false))
       CMusicDatabase::SetPropertiesFromAlbum(*pItem,album);
 
-    path = pItem->GetMusicInfoTag()->GetURL();
+    // since we're going to load additional tag information we
+    // need to replace the library path with the actual file path.
+    // this has to be done before creating the infotag loader.
+    pItem->SetPath(pItem->GetMusicInfoTag()->GetURL());
```

Comment:
Maybe blindly using `CFileItem::GetPath()` is not the best approach and we should first check if the `CMusicInfoTag` of the `CFileItem` has a valid path using `CMusicInfoTag::GetURL()` and only fall back to `CFileItem::GetPath()` if it's empty?